clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -898,6 +837,30 @@ private HeartbeatRequestManager createHeartbeatRequestManager(
                 heartbeatState,
                 heartbeatRequestState,
                 backgroundEventHandler,
-                metrics);
+                new Metrics());
+    }
+
+    private void mockJoiningMemberData(String instanceId) {
```

Comment:
I realize now that we have to properly set the expected state here because it is considered when building the request data, to determine if it should be a full heartbeat or not (HeartbeatRequestManager [ln#566](https://github.com/apache/kafka/blob/f595802cc752ed01dc74e9ab932209fe25a9d10b/clients/src/main/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManager.java#L566)), so I would say that this should include `when(membershipManager.state()).thenReturn(MemberState.JOINING)`

Reply:
I see, good catch! Changed now